packages/core/src/Combobox/ComboboxRoot.vue

```diff
@@ -181,6 +181,10 @@ watch(() => open.value, () => {
   })
 }, { flush: 'post' })
 
+watch(() => allItems.value.size, (val) => {
```

Comment:
I might have spotted this bug as well when a ComboboxItem is mounted mid-search.

Don't you think that we can update L172 as

```ts
watch(() => [filterState.search, allItems.value.size], () => {
```

Reply:
OK I think I found another issue in ComboboxItem a bit related to this - the `props.textValue || currentElement.value.textContent || currentElement.value.innerText` can update for a node while `id` remains same, and in that case it gets outdated.

Reply:
Hi, yes you are right. Thanks for spotting that.